kernel/core/include/os.ads

```diff
@@ -74,28 +74,28 @@ is
    subtype os_virtual_address_t is types.uint32_t;
 
    type os_mbx_entry_t is record
-      sender_id        : aliased os_task_id_t;
-      msg              : aliased os_mbx_msg_t;
+      sender_id        : os_task_id_t;
```

Comment:
if data is shared with C or assembly code, isn't it necessary for them to have an address and therefore to be aliased?

Reply:
aliased is dedicated for Ada world in order to tell the compiler that this location may be aliased by an access type. Well for memory fit with C interface, prefer compiler clauses as layout, size and alignment.
See https://en.wikibooks.org/wiki/Ada_Programming/Keywords/use#Representation_clause.